torchtitan/config_manager.py

```diff
@@ -290,6 +290,23 @@ def __init__(self):
                 Could be implemented as a separate script, but this way shares more code.
             """,
         )
+        self.parser.add_argument(
+            "--checkpoint.async_mode",
+            type=str,
+            default="disabled",
+            help="""
+                Which async checkpoint mode to use. Currently there are 3 different modes.
+                1. "disabled": synchronized checkpointing will be used.
+                2. "async": torch.distributed.checkpoint.async_save will be used.
+                3. "async_with_pinned_mem": a dedicated pinned memory space will be used and
```

Comment:
What is the recommendation to users on how to choose between `"async"` vs. `"async_with_pinned_mem"`?

How should users think about the tradeoffs? Fore xample, is the difference only about performance? Is there a system compatibility constraint? (e.g. how much CPU RAM is required?)

Reply:
Good point, I'll add the memory constraint to the description Using `async` may be enough for most users. `async_with_pinned_mem` should be used when users pursue zero-overhead (almost zero-overhead) checkpoint. Will also add this to the description.